### PR TITLE
ZI LIA: Offset channel and unicode fix

### PR DESCRIPTION
**Pull request by Kuhlen**

### CODE REVIEW

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def performOpen(self, options={}):
             self.device = self.comCfg.address
         
         try:
-            devtype = self.ziConnection.getByte(str('/%s/features/devtype' % self.device))
+            devtype = self.ziConnection.getByte('/%s/features/devtype' % self.device)
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: I'd keep the type casting here as well

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def performOpen(self, options={}):
             return
         
         #Check Options
-        devoptions = self.ziConnection.getByte(str('/%s/features/options' % self.device))
+        devoptions = self.ziConnection.getByte('/%s/features/options' % self.device)
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: I'd keep the type casting here as well

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 3]
```diff
@@ -62,11 +62,12 @@ def performSetValue(self, quant, value, sweepRate=0.0, options={}):
                         ['Mod1On', 'Mod2On'] + \
                         ['Out'+str(x+1)+'SigOut' + str(y+1) + 'On' for x in range(8) for y in range(2)] + \
                         ['Demod'+str(x+1)+'On' for x in range(8)]:
-            self.ziConnection.setInt(str(quant.get_cmd % self.device), 1 if value else 0)
+            self.ziConnection.setInt(str(quant.get_cmd % self.device, 1 if value else 0))
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: str() should end before second argument (after self.device)

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 4]
```diff
@@ -169,14 +171,14 @@ def performGetValue(self, quant, options={}):
                 else:
                     self.wait(self.getValue("TraceStepDelayA"))
                     self.log("Setpoint A: " + str(self.getValue("TraceStepSetpointA")))
-                    self.ziConnection.setDouble(str(self.instrCfg.getQuantity('TraceStepChannel').getCmdStringFromValue(self.getValue("TraceStepChannel")) % self.device), self.getValue("TraceStepSetpointA"))
+                    self.ziConnection.setDouble(self.instrCfg.getQuantity('TraceStepChannel').getCmdStringFromValue(self.getValue("TraceStepChannel")) % self.device, self.getValue("TraceStepSetpointA"))
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: keep typecasting to str for all ziConnection functions

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 5]
```diff
@@ -169,14 +171,14 @@ def performGetValue(self, quant, options={}):
...
                     self.wait(self.getValue("TraceStepDelayB"))
                     self.log("Setpoint B: " + str(self.getValue("TraceStepSetpointB")))
-                    self.ziConnection.setDouble(str(self.instrCfg.getQuantity('TraceStepChannel').getCmdStringFromValue(self.getValue("TraceStepChannel")) % self.device), self.getValue("TraceStepSetpointB"))
+                    self.ziConnection.setDouble(self.instrCfg.getQuantity('TraceStepChannel').getCmdStringFromValue(self.getValue("TraceStepChannel")) % self.device, self.getValue("TraceStepSetpointB"))
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: keep typecasting to str for all ziConnection functions

File: Zurich Instruments UHF/Zurich Instruments UHF.py [hunk 6]
```diff
@@ -169,14 +171,14 @@ def performGetValue(self, quant, options={}):
...
                 while not rec.finished():
                     self.wait(0.05)
                 self.traceBuffer = rec.read(True)
-                self.clockbase = float(self.ziConnection.getInt(str('/%s/clockbase' % self.device)))
+                self.clockbase = float(self.ziConnection.getInt('/%s/clockbase' % self.device))
```
Comment (simongus) [inline Zurich Instruments UHF/Zurich Instruments UHF.py]: keep typecasting to str for all ziConnection functions

**Comment by Kuhlen** [comment]
Hi Simon,

you are right - sorry, should have tested all channels. I am just not
exactly sure about the last one: str('/%s/clockbase' % self.device) - This
is just a string constant, which should not give any trouble - at least it
did not when testing. Do you expect this to become a unicode string in
Python 3 as well?

Cheers,
Sebastian


Simon Gustavsson <notifications@github.com> schrieb am Do., 30. März 2017
um 16:35 Uhr:

**Comment by simongus** [comment]
I’m just trying to cover all bases - if “self.device” comes from the address string in the Labber driver dialog, it’s a unicode string and needs to be typecast under py2.  If self.device comes from the ZI code, then the conversion would not be necessary.  However, it doesn’t really add any overhead to add the typecasting, so it seems better to just add the typecasting everywhere

**Comment by Kuhlen** [comment]
You're right - we used autodetect, so for us it came from ZI.
- Sebastian

Simon Gustavsson <notifications@github.com> schrieb am Do., 30. März 2017
um 16:59 Uhr:

**Pull request merged**